Backend/BikesRentalServer.Tests/ServicesTests/UsersServiceTests/GetAllTechs.cs

```diff
@@ -0,0 +1,101 @@
+﻿using BikesRentalServer.Models;
+using BikesRentalServer.Services;
+using FluentAssertions;
+using System.Collections.Generic;
+using Xunit;
+
+namespace BikesRentalServer.Tests.ServicesTests.UsersServiceTests
+{
+    public class GetAllTechs : UsersServiceTestsBase
+    {
+        [Fact]
+        public void GetAllTechsShouldReturnEmptyListWhenNoTechs()
+        {
+            UsersRepository.Setup(r => r.GetAllTechs()).Returns(new List<User>());
+
+            var usersService = GetUsersService();
+            var result = usersService.GetAllTechs();
+
+            result.Status.Should().Be(Status.Success);
+            result.Object.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void GetAllTechsReturnsAllExistingTechs()
+        {
+            var techs = new[]
+            {
+                new User
+                {
+                    Role = UserRole.Tech,
+                    Status = UserStatus.Active,
+                    Username = "wojtek",
+                },
+                new User
+                {
+                    Role = UserRole.Tech,
+                    Status = UserStatus.Active,
+                    Username = "kasia",
+                },
+                new User
+                {
+                    Role = UserRole.Tech,
+                    Status = UserStatus.Active,
+                    Username = "kacper",
+                },
+            };
+            UsersRepository.Setup(r => r.GetAllTechs()).Returns(techs);
+
+            var usersService = GetUsersService();
+            var result = usersService.GetAllTechs();
+
+            result.Status.Should().Be(Status.Success);
+            result.Object.Should().BeEquivalentTo(techs);
+        }
+
+        [Fact]
+        public void GetAllTechsDontIncludeAdminAndNormalUsers()
+        {
+            var users = new[]
+            {
+                new User
+                {
+                    Role = UserRole.Tech,
+                    Status = UserStatus.Active,
+                    Username = "wojtek",
+                },
+                new User
+                {
+                    Role = UserRole.Tech,
+                    Status = UserStatus.Active,
+                    Username = "kasia",
+                },
+                new User
+                {
+                    Role = UserRole.Admin,
+                    Status = UserStatus.Active,
+                    Username = "wiesiek",
+                },
+                new User
+                {
+                    Role = UserRole.Tech,
+                    Status = UserStatus.Active,
+                    Username = "maniek",
+                },
+                new User
+                {
+                    Role = UserRole.User,
+                    Status = UserStatus.Active,
+                    Username = "kacper",
+                },
+            };
+            UsersRepository.Setup(r => r.GetAllTechs()).Returns(users);
```

Comment:
Komentarz Piotrka mi uświadomił inną rzecz. Ten test jest źle napisany bo to by trzeba dodać tych userów do db, a nie zwracać w repozytorium

Reply:
Nie no, dobrze jest, tylko niepotrzebny ten test xD

Reply:
Nie jest dobrze, bo `GetAllTechs` powinno zwracać samych techów, a nie chmarę różnych userów

Reply:
Jakby był dobrze to by przechodził -,-

Reply:
Ono powinno zwracać to co repo zwraca, ufamy własnej abstrakcji :)

Reply:
No a w tej wersji nie zwraca tego co repo by zwróciło. A że niepotrzebny to druga rzecz